src/vtarget-out/inv-syn/vtarget_gen_inv_abc.cc

```diff
@@ -0,0 +1,618 @@
+  else
+    fout << "echo 'Nothing to check!'" << std::endl;
+} // Export_script
+
+
+/// For jasper, this means do nothing, for yosys, you need to add (*keep*)
+void VlgSglTgtGen_Abc::Export_modify_verilog() {
+  // collect all locations: filename -> lineno
+  // open, read, count and write
+  // if it is a port name, we will ask user to specify its upper level
+  // signal name
+  VerilogModifier vlg_mod(vlg_info_ptr,
+    static_cast<VerilogModifier::port_decl_style_t>(_vtg_config.PortDeclStyle),
+    _vtg_config.CosaAddKeep, supplementary_info.width_info);
+
+  for (auto&& refered_vlg_item : _all_referred_vlg_names) {
+    auto idx = refered_vlg_item.first.find("[");
+    auto removed_range_name = refered_vlg_item.first.substr(0, idx);
+    vlg_mod.RecordKeepSignalName(removed_range_name);
+    // auto sig = // no use, this is too late, vlg_wrapper already exported
+    vlg_mod.RecordConnectSigName(removed_range_name,
+                                 refered_vlg_item.second.range);
+    // vlg_wrapper.add_output(sig.first, sig.second);
+  }
+  vlg_mod.FinishRecording();
+
+
+  //auto tmp_fn = os_portable_append_dir(_output_path, tmp_design_file);
+  auto tmp_fn = os_portable_append_dir(_output_path, top_file_name);
+  // now let's do the job
+  for (auto&& fn : vlg_design_files) {
+    std::ifstream fin(fn);
+    std::ofstream fout(tmp_fn, std::ios_base::app); // append to a temp file
+    if (!fin.is_open()) {
+      ILA_ERROR << "Cannot read file:" << fn;
+      continue;
+    }
+    if (!fout.is_open()) {
+      ILA_ERROR << "Cannot open file for write:" << tmp_fn;
+      continue;
+    }
+    vlg_mod.ReadModifyWrite(fn, fin, fout);
+  } // for (auto && fn : vlg_design_files)
+  
+  // handles the includes
+  // .. (copy all the verilog file in the folder), this has to be os independent
+  if (vlg_include_files_path.size() != 0) {
+    // copy the files and specify the -I commandline to the run.sh
+    for (auto&& include_path : vlg_include_files_path)
+      os_portable_copy_dir(include_path, _output_path);
+  }
+
+} // Export_modify_verilog
+
+
+/// export the memory abstraction (implementation)
+/// Yes, this is also implementation specific, (jasper may use a different one)
+void VlgSglTgtGen_Abc::Export_mem(const std::string& mem_name) {
+  // we will ignore the mem_name
+
+  auto outfn = os_portable_append_dir(_output_path, top_file_name);
+  std::ofstream fout(outfn, std::ios_base::app); // append
+
+  VlgAbsMem::OutputMemFile(fout, _vtg_config.VerificationSettingAvoidIssueStage);
+}
+
+
+static void correct_blif_remove_non_init(const std::string & fn, const std::string & fo) {
+  std::ifstream fin(fn);
+  std::ofstream fout(fo);
+  if (!fin.is_open()) { ILA_ERROR << "Unable to read " << fn; return;}
+  if (!fout.is_open()) { ILA_ERROR << "Unable to write " << fo; return;}
+  std::string line;
+  while(std::getline(fin,line)) {
+    if (line.find(".latch ") == 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Inefficient usage of string::find() in condition; string::starts_with() could be faster.](https://app.codacy.com/app/Bo-Yuan-Huang/ILAng/pullRequest?prid=4555809)